### PR TITLE
Impersonation context added and Form validation added

### PR DESCRIPTION
**Pull request by mahdi13**

### CODE REVIEW

File: restfulpy/principal.py [hunk 1]
```diff
@@ -93,9 +93,12 @@ def id(self):
...
 
-class DummyIdentity:
+class DummyIdentity(JwtPrincipal):
     def __init__(self, *roles):
         self.roles = list(roles)
+        super().__init__({
+            'roles': self.roles
+        })
 
     def is_in_roles(self, *roles):
```
Comment (pylover) [inline restfulpy/principal.py]: I think is not necessary to override this method

File: restfulpy/validation.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                if 'role' not in rule or context.identity.is_in_roles(*rule['role']):
+                    for key, parameters in rule.items():
+                        if key not in effective_rules:
+                            effective_rules[key] = []
+                        effective_rules[key].extend(
+                            [param for param in parameters if param not in effective_rules[key]])
+            effective_rules.pop('role')
+
+            for action, fields in effective_rules.items():
+                locals()[''.join(['_', action])](fields)
```
Comment (pylover) [inline restfulpy/validation.py]: nice_trick, but I recommend you to not use the `**kwargs` and define the arguments, with a name to increase the explicitly.

Comment (pylover) [inline restfulpy/validation.py]: It help's the pycharm too for autocompletion

**Comment by pylover** [comment]
And please squash your commits.

**Comment by mahdi13** [comment]
Thanks.
Changes applied and squashed.

**Pull request closed without merging**